src/services/review/RepoCheckerService.ts

```diff
@@ -316,54 +410,130 @@ export default class RepoCheckerService {
         logger.info(`Site checker for repo ${repo} completed successfully!`)
         return ResultAsync.combineWithAllErrors(findErrors)
           .orElse((e) => {
-            logger.error(`Error running site checker for repo ${repo}: ${e}`)
+            logger.error(
+              `SiteCheckerError: Error running site checker for repo ${repo}: ${e}`
+            )
             // it is ok to have errors, still report the found errors for now
             return okAsync([undefined])
           })
           .andThen(() => okAsync(errors))
       })
   }
 
-  reporter(repo: string, errors: RepoError[]) {
-    // todo: make sure to set up repo in efs in both staging and production environments
+  reporter(
+    repo: string,
+    errors: RepoError[]
+  ): ResultAsync<RepoError[], SiteCheckerError> {
     const folderPath = path.join(SITE_CHECKER_REPO_PATH, repo)
-
     if (!fs.existsSync(folderPath)) {
       fs.mkdirSync(folderPath, { recursive: true })
     }
-    const filePath = path.join(SITE_CHECKER_REPO_PATH, repo, "log.csv")
+    const filePath = path.join(SITE_CHECKER_REPO_PATH, repo, REPO_LOG)
     const stringifiedErrors = this.convertToCSV(errors)
     if (!fs.existsSync(filePath)) {
       fs.appendFileSync(filePath, stringifiedErrors)
     } else {
       fs.writeFileSync(filePath, stringifiedErrors)
     }
-
+    // pushing since there is no real time requirement for user
     return ResultAsync.fromPromise(
       this.git
         .cwd({ path: SITE_CHECKER_REPO_PATH, root: false })
+        .fetch()
+        .merge(["origin/main"])
```

Comment:
What's this merge for? Will there be conflicts?

Reply:
yes, conflicts due to cron job that runs + commits for every push. (https://github.com/isomerpages/isomer-site-checker/)

Reply:
@kishore03109 so how are conflicts resolved? what's the merge strategy? do we need to specify that?